src/renderers/webgl/WebGLProgram.js

```diff
@@ -672,19 +673,20 @@ function WebGLProgram( renderer, cacheKey, parameters, bindingStates ) {
 	vertexShader = unrollLoops( vertexShader );
 	fragmentShader = unrollLoops( fragmentShader );
 
-	if ( parameters.isWebGL2 && ! parameters.isRawShaderMaterial ) {
+	if ( parameters.glslVersion == null && parameters.isWebGL2 && ! parameters.isRawShaderMaterial ) {
```

Comment:
It seems the built-in materials (like `MeshBasicMaterial`) are now not converted anymore.

Reply:
They are converted because `glslVersion == null` for those. This was a part of both of my comments in the other two sections.

Reply:
I don't understand how `parameters.glslVersion` is `null` for `MeshBasicMaterial`. `MeshBasicMaterial` is not derived from `ShaderMaterial`. The `glslVersion` property of `MeshBasicMaterial` is `undefined` (and not `null`).

Reply:
null == undefined, but I can make this `== void(0)` if you want it to be as precise as possible.

Reply:
Oh, I see this now that you have use `==`. Please don't do that. We always use `===`.

Reply:
It would have to test `glslVersion === null && glslVersion === undefined` for that to work, but that's fine if that's the stylistic preference.

Reply:
Yes, this is the preferred style.